src/main/resources/assets/js/DatePickerClear.ts

```diff
@@ -1,52 +1,72 @@
+import { DivEl } from 'lib-admin-ui/dom/DivEl';
+import { Element } from 'lib-admin-ui/dom/Element';
 import { Button } from 'lib-admin-ui/ui/button/Button';
-import { DatePicker, DatePickerBuilder, DatePickerShownEvent } from 'lib-admin-ui/ui/time/DatePicker';
-import { DatePickerPopupBuilder } from 'lib-admin-ui/ui/time/DatePickerPopup';
+import { DatePicker, DatePickerBuilder } from 'lib-admin-ui/ui/time/DatePicker';
+import { DateTimePickerPopup, DateTimePickerPopupBuilder } from 'lib-admin-ui/ui/time/DateTimePickerPopup';
 
 export class DatePickerClear extends DatePicker {
-    private popupClearButton: Button;
-
     constructor(id?: string) {
         super(new DatePickerBuilder());
         if (id) {
             this.getEl().setId(id);
         }
-
-        this.popupClearButton = new Button('Clear');
-        this.popupClearButton.addClass('clear-button');
     }
 
     /**
      * Shamlessly overriding a parent with the same function (ish)
      */
-    protected initPopup() {
-        this.popup = new DatePickerPopupBuilder().setDate(this.selectedDate).build();
+    protected createPopup(): DateTimePickerPopup {
+        const popupBuilder: DateTimePickerPopupBuilder = new DateTimePickerPopupBuilder()
+            .setDate(this.selectedDateTime)
+            .setManageDate(this.builder.manageDate)
+            .setManageTime(this.builder.manageTime);
+
+
+        if (this.builder.timezone) {
+            popupBuilder.setTimezone(this.builder.timezone);
+        }
+
+        if (this.builder.useLocalTimezoneIfNotPresent) {
+            popupBuilder.setUseLocalTimezoneIfNotPresent(true);
+        }
 
-        this.popup.onShown(() => {
-            new DatePickerShownEvent(this).fire();
+        if (this.builder.defaultValue) {
+            popupBuilder.setDefaultValue(this.builder.defaultValue);
+        }
+
+        const clearButton = new Button('Clear');
+        clearButton.addClass('clear-button');
+
+        clearButton.onClicked(() => {
+            this.onClear();
         });
 
-        this.popup.appendChild(this.popupClearButton);
+        return new DatePickerClearPopup(popupBuilder, clearButton);
```

Comment:
Too much code. As it's now possible to override the button, what you need here is to create a popup, then add click handler:

```
protected createPopup(): DateTimePickerPopup {
        const popup: DateTimePickerPopup = new DatePickerClearPopup(new DateTimePickerPopupBuilder().setManageDate(true));
        popup.getSubmitButton().onClicked(() => this.onClear());

        return popup;
    }
```